docs/Building Environments in Aviary.ipynb

```diff
@@ -0,0 +1,434 @@
+    "\n",
+    "    def check_answer(self, answer: str) -> tuple[bool, float, Literal[True]]:\n",
+    "        \"\"\"Check if the proposed answer is correct.\n",
+    "\n",
+    "        Args:\n",
+    "            answer: Proposed answer.\n",
+    "\n",
+    "        Returns:\n",
+    "            Three-tuple of if correct, associated reward (correct_reward if correct,\n",
+    "                tool_failure_reward if tool failure, otherwise incorrect_reward), and\n",
+    "                True indicating done.\n",
+    "        \"\"\"\n",
+    "        try:\n",
+    "            correct: bool = (\n",
+    "                abs(float(answer) - self.answer)\n",
+    "                / (abs(self.answer) + self.config.rel_tol)\n",
+    "                < self.config.rel_tol\n",
+    "            )\n",
+    "            reward = (\n",
+    "                self.config.correct_reward if correct else self.config.incorrect_reward\n",
+    "            )\n",
+    "        except ValueError:\n",
+    "            return False, self.config.tool_failure_reward, True\n",
+    "        else:\n",
+    "            return correct, reward, True\n",
+    "\n",
+    "    def calculator(self, expr: str) -> tuple[float | str | None, float, bool]:\n",
+    "        \"\"\"Calculate a mathematical expression.\n",
+    "\n",
+    "        Args:\n",
+    "            expr: A valid python expression\n",
+    "\n",
+    "        Returns:\n",
+    "            Result of the expression\n",
+    "        \"\"\"\n",
+    "        try:\n",
+    "            expr = expr.strip()\n",
+    "            result = eval(expr)\n",
+    "            with contextlib.suppress(ValueError):\n",
+    "                if int(result) == result:\n",
+    "                    result = int(result)\n",
+    "\n",
+    "        except Exception:\n",
+    "            return (\n",
+    "                \"Error using calculator\",\n",
+    "                self.config.tool_failure_reward,\n",
+    "                self.config.done_on_failure,\n",
+    "            )\n",
+    "        return result, self.config.tool_success_reward, False"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8a7dbf60-9c9c-46e4-bc80-785debabf54d",
+   "metadata": {},
+   "source": [
+    "We next define an instance of `TaskDataset` to load the GSM8K dataset from Hugging Face and provide convenience methods for accessing specific questions, namely `get_new_env_by_idx`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8817f9d3-78e7-47c6-9ba9-1b18bb30edd5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from enum import StrEnum\n",
+    "\n",
+    "import datasets\n",
+    "import pandas as pd\n",
+    "\n",
+    "# SEE: https://huggingface.co/datasets/openai/gsm8k\n",
+    "GSM8K_PUBLIC_SOURCE = \"openai/gsm8k\"\n",
+    "\n",
+    "\n",
+    "class GSM8kDataset(TaskDataset):\n",
+    "    \"\"\"A dataset class for GSM8k.\n",
+    "\n",
+    "    Attributes:\n",
```

Comment:
Same comment from other class here, `Attributes` should go to `__init__` docstring